cmd/check.go

```diff
@@ -152,58 +189,63 @@ func initialiseColorScheme() error {
+func displayControlResults(ctx context.Context, executionTree *execute.ExecutionTree) error {
 	outputFormat := viper.GetString(constants.ArgOutput)
-
-	switch outputFormat {
-	case "text", "brief":
-		err = displayTextOutput(ctx, executionTree)
-	case "csv":
-		err = displayCsvOutput(ctx, executionTree)
-	case "json":
-		err = displayJsonOutput(ctx, executionTree)
-	case "none":
-		// set progress to false
-		viper.Set(constants.ArgProgress, false)
-		// no error, nothing else to do
-	default:
-		err = fmt.Errorf("invalid output format '%s' - must be one of json,csv,text,brief,none", outputFormat)
-	}
-
-	return
-}
-
-func displayCsvOutput(context.Context, *execute.ExecutionTree) error {
-	return fmt.Errorf("CSV output not supported yet")
-}
-
-func displayJsonOutput(ctx context.Context, tree *execute.ExecutionTree) error {
-	bytes, err := json.MarshalIndent(tree.Root, "", "  ")
+	formatter, _ := controldisplay.GetOutputFormatter(outputFormat)
+	formattedReader, err := formatter.Format(ctx, executionTree)
 	if err != nil {
 		return err
 	}
-	fmt.Println(string(bytes))
-	return nil
+	_, err = io.Copy(os.Stdout, formattedReader)
+	return err
 }
 
-func displayTextOutput(ctx context.Context, executionTree *execute.ExecutionTree) error {
-	maxCols := getMaxCols()
-
-	renderer := controldisplay.NewTableRenderer(executionTree, maxCols)
-
-	if ctx.Err() != nil {
-		utils.ShowError(ctx.Err())
+func exportControlResults(ctx context.Context, executionTree *execute.ExecutionTree, formats []controldisplay.CheckExportFormat) error {
+	fmt.Println()
+	for _, format := range formats {
+		fmt.Println("exporting", format.Format, format.File)
+		formatter, err := controldisplay.GetExportFormatter(format.Format)
+		if err != nil {
+			return err
+		}
+		formattedReader, err := formatter.Format(ctx, executionTree)
+		if err != nil {
+			return err
+		}
+		// create the output file
+		destination, err := os.Create(format.File)
+		if err != nil {
+			return err
+		}
+		_, err = io.Copy(destination, formattedReader)
+		if err != nil {
+			return err
+		}
+		destination.Close()
 	}
 
-	fmt.Println(renderer.Render())
 	return nil
 }
 
-func getMaxCols() int {
-	maxCols, _, _ := gows.GetWinSize()
-	// limit to 200
-	if maxCols > 200 {
-		maxCols = 200
+func getExportFormats(executing string) []controldisplay.CheckExportFormat {
+	formats := []controldisplay.CheckExportFormat{}
+	exports := viper.GetStringSlice(constants.ArgExport)
+	for _, export := range exports {
+		parts := strings.SplitN(export, ":", 2)
+		format := parts[0]
+		fileName := generateDefaultExportFileName(format, executing)
+		if len(parts) > 1 {
+			if f, err := helpers.Tildefy(parts[1]); err == nil {
+				fileName = f
+			} else {
+				fileName = parts[1]
+			}
+		}
+		formats = append(formats, controldisplay.NewCheckOutputFormat(format, fileName))
 	}
-	return maxCols
+	return formats
+}
+
+func generateDefaultExportFileName(format string, executing string) string {
```

Comment:
does John specify a file format and does this match it - or have you come up with this yourself?